knora/dsplib/models/helpers.py

```diff
@@ -180,14 +180,14 @@ def __init__(self, context: Optional[Dict[str, str]] = None):
                 if self._context.get(cc[0]) is None:
                     self._context[cc[0]] = cc[1]
         else:
```

Comment:
This else branch duplicates the default settings already checked in line 171-181. Would it make sense to move the checks in line 171-181 outside the if-branch, so that it will be checked for any given context (and remove the else branch instead)? Like:

```
self._context = ContextType({})

if context:
    <snip>
    for prefix, onto in cleaned_input.items():
        self._context[prefix] = OntoInfo(onto[:-1], True) if onto.endswith('#') else OntoInfo(onto, False)

# we always want the base ontologies/prefixes included in the context
#
for cc in self.base_ontologies.items():
    if self._context.get(cc[0]) is None:
        self._context[cc[0]] = cc[1]

#
# we always want the knora ontologies/prefixes included in the context
#
for cc in self.knora_ontologies.items():
    if self._context.get(cc[0]) is None:
        self._context[cc[0]] = cc[1]
```

Reply:
you're absolutely right, will change that. Thanks! (I tried to focus [with limited success] on the issue actually addressed in this PR - I'm sure there is a near infinite amount of stuff that could be cleaned up...)